core/indexed/indexed.go

```diff
@@ -0,0 +1,40 @@
+package indexed
+
+import (
+	"bytes"
+	"iter"
+
+	"github.com/NethermindEth/juno/encoder"
+)
+
+type BufferedEncoder struct {
+	*bytes.Buffer
+	encoder.Encoder
+}
+
+func NewBufferedEncoder() BufferedEncoder {
+	var buf bytes.Buffer
+	return BufferedEncoder{
+		Buffer:  &buf,
+		Encoder: encoder.NewEncoder(&buf),
+	}
+}
+
+func Write(
+	writer BufferedEncoder,
+	items iter.Seq2[any, error],
+) ([]int, error) {
+	indexes := make([]int, 0)
```

Comment:
Is there anyway for us to know the size of the elements of the Buffer in advance, so that we can pre-plan and build the `indexes` array with the right size. 

Maybe this is not true because same type of elements can have different sizes when serialized and hence is not possible.